lint.sh

```diff
@@ -31,14 +31,16 @@ if [[ -z "${PWNDBG_VENV_PATH}" ]]; then
     PWNDBG_VENV_PATH="./.venv"
 fi
 
-source "${PWNDBG_VENV_PATH}/bin/activate"
+if [[ "${PWNDBG_VENV_PATH}" != "PWNDBG_PLEASE_SKIP_VENV" ]]; then
+    source "${PWNDBG_VENV_PATH}/bin/activate"
+fi
 
 set -o xtrace
 
 LINT_FILES="pwndbg tests *.py"
 LINT_TOOLS="isort ruff vermin mypy"
 
-if ! type ${LINT_TOOLS} &> /dev/null; then
+if ! type ${LINT_TOOLS} &>/dev/null; then
```

Comment:
The linter is complaining about this change

Reply:
Sorry, my editor's linters re-changes every time I save the file, and I guess forget to run your linter before final commit.

Not sure it's been discussed, but it may be worth considering using something like [pre-commit](https://pre-commit.com/) so that it hooks the commit and forces the linting then, rather than having to remember to run lint.sh every time.

Reply:
Should be fixed now

Reply:
Regarding pre-commit, the setup-dev.sh script asks if you want to add a git hook that lints before pushing. Seems like pre-commit is also just run through a git hook, so not sure if there's benefit in switching to it, but I'm open to it if there's a compelling reason.

Reply:
Ah makes sense. Since setup-dev.sh doesn't work on nix, I didn't run it fully and so didn't see the hook question. I will look into installing the hooks in the nix dev shell I added though.

To me the extra benefits of something like pre-commit is you get installation and version pinning for all linters, not just python stuff. 

For example shfmt comes from various OS package managers in setup-dev.sh, and so everyone could be running different versions. pre-commit would install the same version of that too (not just whatever python tooling is available via poetry file). Could also be used to more easily enforce additional linting like whitespacing in documentation, markdown structure consistency across the project etc. 

Also you wouldn't need to re-run setup-dev.sh if something new is added. the pre-commit stuff would automatically have everyone using the same linters and versions.

Anyway, up to you folks and maybe not needed. I've used it on a few projects and seems nice though, so figure I'd mention.